src/app/campaign/store/selectors/campaign-flight.selectors.ts

```diff
@@ -69,3 +76,85 @@ export const selectRoutedCampaignFlightDocs = createSelector(
   }
 );
+
+export const selectCampaignFlightInventoryReportData = createSelector(
+  selectCampaign,
+  selectAllInventory,
+  selectAllAdvertisers,
+  selectRemoteFlightsOrderByContractStartAt,
+  selectFlightDaysEntities,
+  (campaign: CampaignState, shows: Inventory[], advertisers: Advertiser[], flights: Flight[], inventory: Dictionary<FlightDays>) => {
+    // Flight Names
+    const reportData: any[][] = [['Flight Name'].concat(flights.map(flight => flight.name))];
+    // Show
+    reportData.push(
+      ['Show'].concat(
+        flights.map(flight => {
+          const show = shows.find(i => i.self_uri === flight.set_inventory_uri);
+          return show ? show.podcastTitle : '';
+        })
+      )
+    );
+    // Advertiser/Sponsor
+    reportData.push(
+      ['Sponsor'].concat(
+        flights.map(_ => {
+          const advertiser = advertisers.find(a => a.set_advertiser_uri === campaign.remoteCampaign.set_advertiser_uri);
+          return advertiser ? advertiser.name : '';
+        })
+      )
+    );
+    // Zones
+    reportData.push(['Zones'].concat(flights.map(flight => flight.zones.map(zone => zone.label).join('|'))));
+    // Contract Start and End Dates
+    reportData.push(
+      ['Start Date'].concat(flights.map(flight => (flight.contractStartAt ? flight.contractStartAt.format('MM-DD-YYYY') : '')))
+    );
+    reportData.push(
+      ['End Date'].concat(flights.map(flight => (flight.contractEndAtFudged ? flight.contractEndAtFudged.format('MM-DD-YYYY') : '')))
+    );
+    // Geo Targets
+    const geoTransform = new GeoTargetsPipe();
+    reportData.push(
+      ['Geotarget'].concat(
+        flights.map(flight =>
+          flight.targets
+            .map(t => geoTransform.transform(t))
+            .filter(t => t)
+            .join('|')
+        )
+      )
+    );
+    // Contract Goal
+    reportData.push((['Contract Goal'] as (string | number)[]).concat(flights.map(flight => flight.contractGoal)));
+    // Actuals
+    reportData.push((['Total Delivered'] as (string | number)[]).concat(flights.map(flight => flight.actualCount)));
+    // Inventory Days for each flight mapped by date
+    const inventoryDays = flights.reduce(
+      (days, flight) =>
+        inventory[flight.id].days.reduce((acc, day) => {
+          const date = day.date.toISOString().slice(0, 10);
+          return {
+            ...acc,
+            [date]: {
+              ...acc[date],
+              [flight.id]: day
+            }
+          };
+        }, days),
+      {}
+    );
+    // push each inventory date row
+    // if no data for flight on date, show '-'
+    Object.keys(inventoryDays)
+      .sort((a, b) => new Date(a).valueOf() - new Date(b).valueOf())
+      .forEach(date =>
+        reportData.push(
+          ([utc(date).format('MM-DD-YYYY')] as (string | number)[]).concat(
+            flights.map(flight => (inventoryDays[date][flight.id] ? inventoryDays[date][flight.id].numbers.actuals : '-'))
```

Comment:
I don't love that in-flight-range days show up as blanks in the CSV instead of 0s.  But I can't remember what ad-ops actually said about that ... so I'm fine with just getting this out there for some feedbacks!

Reply:
In the API, dates with no actuals are null, that's why you're seeing them as blanks instead of '-' where there is no expected data. I think they are expecting there to be actuals during the flight dates, so I'm not actually sure what they want when there isn't or if the flight hasn't finished. Yeah, feedback would be good.

Reply:
Ah, okay.  I do remember them wanting dashes.  To me I guess `0` is more indicative of "this flight _could have_ served this day, but we got 0 actuals".  And `-` sort of means "not available".  But we can get feedback on that!

Reply:
If it _could_ have served, like if the date is in the past and there are still no actuals, the API probably really should have zero instead of null. null makes sense to me for dates in the future, or if they really do want '-' filled in there because they are exporting the report before the end of the flight. There is no differentiation of when the flight should be running or not if we just fill everything empty with '-'. But hey, let's not decide requirements in the PR, not the right place.